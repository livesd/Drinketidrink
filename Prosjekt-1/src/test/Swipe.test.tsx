import "@testing-library/jest-dom/vitest";
import { render } from "@testing-library/react";
import { vi } from "vitest";
import CocktailCard from "../components/Swipe";
import type { Drink } from "../api/cocktails";

// Mock the toIngredients function
vi.mock("../api/cocktails", () => ({
  toIngredients: vi.fn(() => [
    { name: "Light rum", measure: "2-3 oz" },
    { name: "Lime", measure: "Juice of 1" },
    { name: "Sugar", measure: "2 tsp" },
    { name: "Mint", measure: "2-4" },
  ])
}));

const mockDrink: Drink = {
  idDrink: "11000",
  strDrink: "Mojito",
  strDrinkThumb: "https://www.thecocktaildb.com/images/media/drink/metwgh1606770327.jpg",
  strCategory: "Cocktail",
  strAlcoholic: "Alcoholic",
  strGlass: "Highball glass",
  strInstructions: "Muddle mint leaves with sugar and lime juice. Add a splash of soda water and fill the glass with cracked ice.",
  strIngredient1: "Light rum",
  strIngredient2: "Lime",
  strIngredient3: "Sugar",
  strIngredient4: "Mint",
  strIngredient5: "Soda water",
  strMeasure1: "2-3 oz",
  strMeasure2: "Juice of 1",
  strMeasure3: "2 tsp",
  strMeasure4: "2-4",
  strMeasure5: null,
};

test("CocktailCard matches snapshot", () => {
  const { container } = render(<CocktailCard drink={mockDrink} />);
  expect(container.firstChild).toMatchSnapshot();
});

test("CocktailCard with minimal data matches snapshot", () => {
  const minimalDrink: Drink = {
    ...mockDrink,
    strAlcoholic: null,
    strGlass: null,
    strInstructions: null,
    strIngredient1: null,
    strIngredient2: null,
    strIngredient3: null,
    strIngredient4: null,
    strIngredient5: null,
  };

  const { container } = render(<CocktailCard drink={minimalDrink} />);
  expect(container.firstChild).toMatchSnapshot();
});