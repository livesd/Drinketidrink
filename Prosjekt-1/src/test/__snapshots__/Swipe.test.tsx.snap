// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CocktailCard matches snapshot 1`] = `
<article
  aria-labelledby="drink-11000-title"
  class="card drink-card"
>
  <header
    class="lyrics-header"
  >
    <h2
      id="drink-11000-title"
    >
      Mojito
    </h2>
  </header>
  <p
    class="meta"
  >
    <strong>
      Alcohol:
    </strong>
     
    Alcoholic
     ·
     
    <strong>
      Glass:
    </strong>
     
    Highball glass
  </p>
  <div
    class="media"
  >
    <img
      alt="Mojito"
      class="thumb"
      decoding="async"
      height="160"
      loading="lazy"
      src="https://www.thecocktaildb.com/images/media/drink/metwgh1606770327.jpg"
      width="160"
    />
    <div>
      <h3
        class="sr-only"
      >
        Ingredients
      </h3>
      <table
        aria-label="Ingredients"
        class="nutrition"
      >
        <tbody>
          <tr>
            <th
              scope="row"
            />
            <td>
              2-3 oz
            </td>
          </tr>
          <tr>
            <th
              scope="row"
            />
            <td>
              Juice of 1
            </td>
          </tr>
          <tr>
            <th
              scope="row"
            />
            <td>
              2 tsp
            </td>
          </tr>
          <tr>
            <th
              scope="row"
            />
            <td>
              2-4
            </td>
          </tr>
        </tbody>
      </table>
      <p
        class="instructions"
      >
        <strong>
          Instructions:
        </strong>
         
        Muddle mint leaves with sugar and lime juice. Add a splash of soda water and fill the glass with cracked ice.
      </p>
    </div>
  </div>
</article>
`;

exports[`CocktailCard with minimal data matches snapshot 1`] = `
<article
  aria-labelledby="drink-11000-title"
  class="card drink-card"
>
  <header
    class="lyrics-header"
  >
    <h2
      id="drink-11000-title"
    >
      Mojito
    </h2>
  </header>
  <p
    class="meta"
  >
    <strong>
      Alcohol:
    </strong>
     
    –
     ·
     
    <strong>
      Glass:
    </strong>
     
    –
  </p>
  <div
    class="media"
  >
    <img
      alt="Mojito"
      class="thumb"
      decoding="async"
      height="160"
      loading="lazy"
      src="https://www.thecocktaildb.com/images/media/drink/metwgh1606770327.jpg"
      width="160"
    />
    <div>
      <h3
        class="sr-only"
      >
        Ingredients
      </h3>
      <table
        aria-label="Ingredients"
        class="nutrition"
      >
        <tbody>
          <tr>
            <th
              scope="row"
            />
            <td>
              2-3 oz
            </td>
          </tr>
          <tr>
            <th
              scope="row"
            />
            <td>
              Juice of 1
            </td>
          </tr>
          <tr>
            <th
              scope="row"
            />
            <td>
              2 tsp
            </td>
          </tr>
          <tr>
            <th
              scope="row"
            />
            <td>
              2-4
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</article>
`;
